server/src/features/users/user.controller.js

```diff
@@ -128,4 +130,195 @@ export default class UserController {
+          message: 'Only students, staff, TAs, and professors can add events to favorites',
+        });
+      }
+
+      // Check if event exists
+      const event = await mongoose.model('Event').findById(eventId);
+      if (!event) {
+        return res.status(404).json({
+          success: false,
+          message: 'Event not found',
+        });
+      }
+
+      // Add to favorites if not already added
+      const user = await User.findById(userId);
+      if (!user.favoriteEvents.includes(eventId)) {
+        user.favoriteEvents.push(eventId);
+        await user.save();
+      }
+
+      return res.status(200).json({
+        success: true,
+        message: 'Event added to favorites',
+        data: {
+          eventId,
+          favoritesCount: user.favoriteEvents.length,
+        },
+      });
+    } catch (err) {
+      next(err);
+    }
+  }
+
+  /**
+   * Remove an event from user's favorites
+   * DELETE /api/users/favorites/:eventId
+   */
+  static async removeFromFavorites(req, res, next) {
+    try {
+      const { eventId } = req.params;
+      const userId = req.user._id;
+
+      // Validate request
+      const { error } = favoriteEventSchema.validate({ eventId });
+      if (error) {
+        return res.status(400).json({
+          success: false,
+          message: error.details[0].message,
+        });
+      }
+
+      // Check if user is allowed to remove from favorites
+      const allowedRoles = ['student', 'staff', 'ta', 'professor'];
+      if (!allowedRoles.includes(req.user.role)) {
+        return res.status(403).json({
+          success: false,
+          message: 'Only students, staff, TAs, and professors can manage favorites',
+        });
+      }
+
+      // Remove from favorites
+      const user = await User.findByIdAndUpdate(
+        userId,
+        { $pull: { favoriteEvents: eventId } },
+        { new: true }
+      );
+
+      if (!user) {
+        return res.status(404).json({
+          success: false,
+          message: 'User not found',
+        });
+      }
+
+      return res.status(200).json({
+        success: true,
+        message: 'Event removed from favorites',
+        data: {
+          eventId,
+          favoritesCount: user.favoriteEvents.length,
+        },
+      });
+    } catch (err) {
+      next(err);
+    }
+  }
+
+  /**
+   * Get user's favorite events
+   * GET /api/users/favorites
+   */
+  static async getFavoriteEvents(req, res, next) {
+    try {
+      const userId = req.user._id;
+
+      // Check if user is allowed to view favorites
+      const allowedRoles = ['student', 'staff', 'ta', 'professor'];
+      if (!allowedRoles.includes(req.user.role)) {
+        return res.status(403).json({
+          success: false,
+          message: 'Only students, staff, TAs, and professors can view favorite events',
+        });
+      }
+
+      // Get user with populated favorite events
+      const user = await User.findById(userId).populate({
```

Comment:
```suggestion
      User.findById(userId, { favoriteEvents: 1 })
  .populate(...)

```

This loads the entire user (email, passwordHash, misc fields) even though you only need favorites.

Reply:
and  use .lean() on the populated query

Reply:
ya sety msh far2a bs done 😂